cpp/include/cudf/stream_compaction.hpp

```diff
@@ -277,6 +277,28 @@ std::unique_ptr<table> distinct(
   nan_equality nans_equal             = nan_equality::ALL_EQUAL,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
+/**
+ * @brief Create a column of indices of all distinct rows in the input table.
+ *
+ * Given an `input` table_view, an output vector of all row indices of the distinct rows is
+ * generated. If there are duplicate rows, which index is kept depends on the `keep` parameter.
+ *
+ * @param input The input table
+ * @param keep Get index of any, first, last, or none of the found duplicates
+ * @param nulls_equal Flag to specify whether null elements should be considered as equal
+ * @param nans_equal Flag to specify whether NaN elements should be considered as equal
+ * @param stream CUDA stream used for device memory operations and kernel launches
+ * @param mr Device memory resource used to allocate the returned vector
+ * @return Column containing the result indices
+ */
+std::unique_ptr<column> distinct_indices(
```

Comment:
I decided not to call it `cudf::get_distinct_indices` because AFAICT nothing is that returns a result is called `get_XXX`.

Reply:
Good call. Let's rename the detail API to match, though (see other comment).